src/daft-io/src/object_store_glob.rs

```diff
@@ -309,6 +309,15 @@ fn _should_return(fm: &FileMetadata) -> bool {
         {
             false
         }
+        // These are marker files used by Spark that we don't want to return
+        FileType::File
+            if fm.filepath.ends_with("_metadata")
```

Comment:
Let's do a `.lower()` on the file names and have these stored as constants?

Ideally it should also be configurable by the user but I think we can ignore that for now.